crypto/evp/keymgmt_meth.c

```diff
@@ -261,8 +261,13 @@ static void *keymgmt_from_algorithm(int name_id,
         return NULL;
     }
     keymgmt->prov = prov;
-    if (prov != NULL)
-        ossl_provider_up_ref(prov);
+    if (prov != NULL) {
+        if (!ossl_provider_up_ref(prov)) {
+            EVP_KEYMGMT_free(keymgmt);
+            ERR_raise(ERR_LIB_EVP, EVP_R_INVALID_PROVIDER_FUNCTIONS);
+            return NULL;
+        }
+    }
```

Comment:
This is a little neater IMO:

```suggestion
    if (prov != NULL && !ossl_provider_up_ref(prov)) {
        EVP_KEYMGMT_free(keymgmt);
        ERR_raise(ERR_LIB_EVP, EVP_R_INVALID_PROVIDER_FUNCTIONS);
        return NULL;
    }
```